examples/titanic_example/assets/algo_random_forest/titanic_algo_rf.py

```diff
@@ -7,167 +7,170 @@
+    # extract titles
+    X["Title"] = X.Name.str.extract(r" ([A-Za-z]+)\.", expand=False)
+    # replace titles with a more common title or as Rare
+    X["Title"] = X["Title"].replace(
+        [
+            "Lady",
+            "Countess",
+            "Capt",
+            "Col",
+            "Don",
+            "Dr",
+            "Major",
+            "Rev",
+            "Sir",
+            "Jonkheer",
+            "Dona",
+        ],
+        "Rare",
+    )
+    X["Title"] = X["Title"].replace("Mlle", "Miss")
+    X["Title"] = X["Title"].replace("Ms", "Miss")
+    X["Title"] = X["Title"].replace("Mme", "Mrs")
+    # convert titles into numbers
+    X["Title"] = X["Title"].map(titles)
+    # filling NaN with 0, to get safe
+    X["Title"] = X["Title"].fillna(0)
+    X = X.drop(["Name"], axis=1)
+
+    # Sex
+    genders = {"male": 0, "female": 1}
+    X["Sex"] = X["Sex"].map(genders)
+
+    # Ticket
+    X = X.drop(["Ticket"], axis=1)
+
+    # Drop non relevant features
+    X = X.drop("not_alone", axis=1)
+    X = X.drop("Parch", axis=1)
+
+    return X
+
+
+def _predict_pandas(model, X):
+    y_pred = model.predict(X)
+    return pd.DataFrame(columns=["Survived"], data=y_pred)
+
+
+def train(inputs, outputs, task_properties):
+
+    X = inputs["datasamples"].drop(columns="Survived")
+    y = inputs["datasamples"].Survived
+    X = _normalize_X(X)
+
+    # the following RFC hyperparameters were determined using:
+    # >>> param_grid = {"criterion": ["gini", "entropy"],
+    #                   "min_samples_leaf": [1, 5, 10, 25, 50, 70],
+    #                   "min_samples_split": [2, 4, 10, 12, 16, 18, 25, 35],
+    #                   "n_estimators": [100, 400, 700, 1000, 1500]}
+    # >>> rf = RandomForestClassifier(n_estimators=100, max_features='auto', oob_score=True,
+    #                                 random_state=1, n_jobs=-1)
+    # >>>,clf = GridSearchCV(estimator=rf, param_grid=param_grid, n_jobs=-1)
+
+    # Random Forest
+    random_forest = RandomForestClassifier(
+        criterion="gini",
+        min_samples_leaf=1,
+        min_samples_split=10,
+        n_estimators=100,
+        oob_score=True,
+        random_state=1,
+        n_jobs=-1,
+    )
+    random_forest.fit(X, y.values.ravel())
+
+    save_model(random_forest, outputs["model"])
+
+
+def predict(inputs, outputs, task_properties):
+    X = inputs["datasamples"].drop(columns="Survived")
+    model = load_model(inputs["models"])
+    X = _normalize_X(X)
+    pred = _predict_pandas(model, X)
+
+    save_predictions(pred, outputs["predictions"])
+
+
+def load_model(path):
+    with open(path, "rb") as f:
+        return pickle.load(f)
+
+
+def save_model(model, path):
+    with open(path, "wb") as f:
+        pickle.dump(model, f)
+
+
+def save_predictions(y_pred, path):
+    with open(path, "w") as f:
+        y_pred.to_csv(f, index=False)
 
 
 if __name__ == "__main__":
-    tools.algo.execute(Algo())
+    tools.execute(train, predict)
```

Comment:
What do we need that now that we specify the entrypoint?

Reply:
We need to pass the functions we want to register to substratools. 
A new interface will be proposed very soon (like in the afternoon) to use decorator to register the functions, and have only `tools.execute()` without arguments